client/app/dbaas/logs/detail/streams/alerts/add/alerts-add.html

```diff
@@ -1,5 +1,195 @@
+                </div>
+            </div>
+
+            <div class="oui-field" data-ng-class="{'oui-field_error': ctrl.form.$submitted && ctrl.form.threshold.$invalid}">
+                <label for="threshold" class="oui-field__label oui-label" data-translate="add_alert_threshold_field"></label>
+                <div class="oui-field__content">
+                    <oui-numeric
+                        id="threshold"
+                        name="threshold"
+                        model="ctrl.alert.threshold"
+                        min="0"
+                        max="99"
+                        required>
+                    </oui-numeric>
+                    <div class="oui-field-helper" data-translate="add_alert_threshold_field_description"></div>
+                </div>
+            </div>
+
+            <div data-ng-if="ctrl.alertType === 'FIELD_VALUE'" class="oui-field">
+                <label for="constraint_type" class="oui-label" data-translate="add_alert_constraint_type_field"></label>
+                <div class="oui-field__content">
+                    <div class="oui-field-control oui-field-control_4">
+                        <div class="oui-select">
+                            <select class="oui-select__input"
+                                name="constraint_type"
+                                id="constraint_type"
+                                data-ng-model="ctrl.alert.constraintType">
+                                <option value="MEAN" data-translate="add_alert_constraint_type_mean"></option>
+                                <option value="MIN" data-translate="add_alert_constraint_type_min"></option>
+                                <option value="MAX" data-translate="add_alert_constraint_type_max"></option>
+                                <option value="SUM" data-translate="add_alert_constraint_type_sum"></option>
+                                <option value="STDDEV" data-translate="add_alert_constraint_type_sd"></option>
+                            </select>
+                            <i class="oui-icon oui-icon-chevron-down" aria-hidden="true"></i>
+                        </div>
+                    </div>
+                    <div class="oui-field-helper" data-translate="add_alert_constraint_type_field_description"></div>
+                </div>
+            </div>
+        </div>
+
+        <div class="oui-field" data-ng-class="{'oui-field_error': ctrl.form.$submitted && ctrl.form.grace_period.$invalid}">
+            <label for="grace_period" class="oui-field__label oui-label" data-translate="add_alert_grace_period_field"></label>
+            <div class="oui-field__content">
+                <oui-numeric
+                    id="grace_period"
+                    name="grace_period"
+                    model="ctrl.alert.grace"
+                    min="1"
+                    max="60"
```

Comment:
For maintainability purpose, you should be the number in a constant and name the constant in a way that we know what it is. maybe "MAX_GRACE_PERIOD_IN_SECONDS" ?

Reply:
There are many in this file. You could put them all in one .constant file.

Reply:
A new constant file has been created and the hard-coded values have been replaced by the constants' values